lib/rubygems/spec_fetcher.rb

```diff
@@ -197,6 +197,24 @@ def suggest_gems_from_name(gem_name, type = :latest)
     matches.first(5).map { |name, dist| name }
   end
 
+  def suggest_gems_from_name_regex(gem_name, type = :latest)
```

Comment:
I noticed this is almost the same as `Gem::SpecFetcher.fetcher.suggest_gems_from_name` but without the `levenshtein_distance` filter. I'm not sure if this is what we want, in theory the `levenshtein_distance` should give us the closest match but I checked this functionality and I see the regex results are useful too. Maybe we can include the two results? 🤔

Reply:
That seems good to me. How do you envision the 2 results getting combined? Currently, it looks like the top 5 results are taken from the levenshtein_distance filter and all the results are returned form the regex filter. I think mashing the two results together might be ok. That might also result in having a really big list of suggestions.

Reply:
Maybe we could show the 10 first results and show first the regex matches that match the name of the gem at the start and then the `levenshtein_distance filetered` ones ?

Reply:
Gave this a crack. Now it will return 10 results. It will try to fill that 10 with the regex filter first, then fill in the rest with the levenshtein filter